elements/lisk-transactions/README.md

```diff
@@ -2,6 +2,91 @@
+The application-specific business logic for custom transaction types is defined according to an abstract [interface](#interface) that is common across all transaction types.
+
+All of the default transaction types of the Lisk SDK transactions implement the abstract interface of the base transaction, and therefore can be used as a role model for custom transactions.
+It's also possible to inherit from one of the default transaction types, in order to extent or modify them.
+
+The default transaction types each implement a different use-case of the Lisk network, i.e:
+
+0. Balance transfer (type 0),
+1. Second signature registration (type 1)
+1. Delegate registration (type 2)
+1. Delegate vote (type 3)
+1. Multisignature account registration (type 4)
+
+> The first 10 transaction types are reserved for the [Lisk protocol](https://lisk.io/documentation/lisk-protocol), don't use them to register custom transactions.
+
+For a complete list of all default transaction types, check out the section [Lisk Transactions](https://lisk.io/documentation/lisk-protocol/transactions) of the Lisk Protocol.
+
+Check out the Lisk SDK [Example Apps](https://github.com/LiskHQ/lisk-sdk-test-app) for simple [code examples of custom transaction types](https://github.com/LiskHQ/lisk-sdk-test-app/blob/development/hello_world/hello_transaction.js).
+
+### Lifecycle
+
+The lifecycle of a transaction in general in Lisk SDK can be summarized as follows:
+
+1. A transaction is created and signed (off-chain). The script to do it is in `src/create_and_sign.ts`.
+2. The transaction is sent to a network. This can be done by a third party tool (like `curl` or `Postman`), but also using Lisk Commander.
+   To send the default transactions for the protocol network, you can rely on the wallet Lisk UI tools - Lisk Hub and Mobile. All of the tools need to be authorized to access an HTTP API of a network node.
+3. A network node receives a transaction and after a lightweight schema validation, adds it to a transaction pool.
+4. In the transaction pool, the transactions are firstly `validated`. In this step, only static checks are performed. These include schema validation and signature validation.
+5. Validated transactions go to the `prepare` step defined in the transaction class, which to limit the I/O database operations prepares all the information relevant to properly `apply` or `undo` the transaction. The store with the prepared data is a parameter of the mentioned methods.
+6. Delegates are forging the valid transactions into the blocks and broadcasting the blocks to the network. Each network node performs the `apply` and `applyAsset` steps after the successful `validate` step.
+7. It is probable, especially shortly after a block is applied, that due to the decentralized network conditions a node does the `undo` step and the block containing all of the included transactions get reverted in favour of a competing block.
```

Comment:
6a.  Based on workflow executing `apply` and `applyAsset` function, the state in memory is either saved to the blockchain or discarded. For instance, if the transaction is being executed within the process of saving new a block in the blockchain, the changes in the memory are saved in the database. In the other case, when the transaction is executed within the domain of the transaction pool, the changes of state in memory are discarded.

Reply:
Added a paragraph 👍